superset/utils/oauth2.py

```diff
@@ -192,3 +192,4 @@ class OAuth2ClientConfigSchema(Schema):
     )
     authorization_request_uri = fields.String(required=True)
     token_request_uri = fields.String(required=True)
+    project_id = fields.String(required=False)
```

Comment:
BigQuery takes `project_id` as a param in its `OAuth2` parameters

Reply:
I think ideally we'd want to store the project id in the URI, to make it compatible with non-oauth2 use cases. And then later when you need you can grab it from `database.sqlalchemy_uri.database`.